sys/net/gnrc/network_layer/ndp2/gnrc_ndp2.c

```diff
@@ -539,22 +539,18 @@ static gnrc_pktsnip_t *_build_headers(gnrc_ipv6_netif_t *netif,
 static size_t _get_l2src(gnrc_ipv6_netif_t *netif, uint8_t *l2src,
                          size_t l2src_maxlen)
 {
-    bool try_long = false;
-    int res;
     uint16_t l2src_len;
-    /* maximum address length that fits into a minimum length (8) S/TL2A option */
+    /* maximum address length that fits into a minimum length (8) S/TL2A
+     * option */
     const uint16_t max_short_len = 6;
 
-    /* try getting source address */
-    if ((gnrc_netapi_get(netif->pid, NETOPT_SRC_LEN, 0, &l2src_len,
-                         sizeof(l2src_len)) >= 0) &&
-        (l2src_len > max_short_len)) {
-        try_long = true;
-    }
-
-    if (try_long && ((res = gnrc_netapi_get(netif->pid, NETOPT_ADDRESS_LONG, 0,
-                                            l2src,
-                                            l2src_maxlen)) > max_short_len)) {
+    /* get source address length */
+    int res = gnrc_netapi_get(netif->pid, NETOPT_SRC_LEN, 0,
+                              &l2src_len, sizeof(l2src_len));
+    /* get source address, try long address first if possible  */
+    if ((res >= 0) && (l2src_len > max_short_len) &&
+        (gnrc_netapi_get(netif->pid, NETOPT_ADDRESS_LONG, 0,
```

Comment:
Same here you changed the behavior, the previous code, set res to the result of gnrc_netapi_get(iface, NETOPT_ADDRESS_LONG,..), here it gets the result of gnrc_netapi_get(iface, NETOPT_SRC_LEN,..).

Reply:
seems to be copy-pasta from above - so my answer applies here too